### PR TITLE
Add nsmgr-proxy into chains

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin <denis.tingajkin@xored.com>

## Motivation
https://github.com/networkservicemesh/sdk/pull/465
This PR is the last part of implementing interdomain in SDK.

This PR fixes some small issues that have been found during testing + adds nsmgr-proxy chain.

### CODE REVIEW

File: pkg/networkservice/chains/nsmgrproxy/server.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+type NSMgrProxy interface {
+	networkservice.NetworkServiceServer
+	Register(s *grpc.Server)
+}
+
+type nsmgrProxy struct {
+	endpoint.Endpoint
+}
+
+func (n *nsmgrProxy) Register(s *grpc.Server) {
```
Comment (edwarnicke) [inline pkg/networkservice/chains/nsmgrproxy/server.go:47]: Is this needed?  Can it be inherited from the embedded endpoint.Endpoint ?

File: pkg/networkservice/chains/nsmgrproxy/server.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/connect"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/interdomainurl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/swap"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/token"
+)
+
+// NSMgrProxy is extended networkservice.NetworkServiceServer interface with additional method Register
+type NSMgrProxy interface {
```
Comment (edwarnicke) [inline pkg/networkservice/chains/nsmgrproxy/server.go:38]: Is this needed?  Can it be inherited from the embedded endpoint.Endpoint?

File: pkg/networkservice/chains/nsmgrproxy/server.go (hunk 1 above)
Comment (denis-tingajkin) [reply pkg/networkservice/chains/nsmgrproxy/server.go]: You are correct, we can use just `endpoint.Endpoint`

File: pkg/networkservice/chains/nsmgrproxy/server.go (hunk 2 above)
Comment (denis-tingajkin) [reply pkg/networkservice/chains/nsmgrproxy/server.go]: Thanks, fixed!

File: pkg/networkservice/chains/nsmgrproxy/server_test.go [hunk 3]
```diff
@@ -26,13 +26,15 @@ import (
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
 	"github.com/networkservicemesh/api/pkg/api/registry"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
 
 	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
 )
 
 func TestNSMGR_InterdomainUseCase(t *testing.T) {
```
Comment (denis-tingajkin) [inline pkg/networkservice/chains/nsmgrproxy/server_test.go:34]: Funny fact, This test testing the logic of interdomain use case with real grpc NSM services and it runs in 120 ms on my machine.

Comment (edwarnicke) [reply pkg/networkservice/chains/nsmgrproxy/server_test.go:34]: :) :) :)

**Pull request merged**